src/com_interface.rs

```diff
@@ -6,24 +6,35 @@ use crate::*;
 /// Implementors of this trait must be transparent wrappers
 /// around pointers to the associated VTable for the interface.
 /// In other words, implementators must be exactly equivalent to
-/// *const *const Self::VTable.
+/// *mut *mut Self::VTable.
 ///
 /// VTables must be COM compliant VTables where the first three function
 /// pointers are the IUknown methods.
+///
+/// Additionally, because ComInterfaces are just pointers to vtables,
+/// it must be safe to zero initialize the interface.
 pub unsafe trait ComInterface: Sized {
-    const GUID: Guid;
+    const IID: Guid;
     type VTable;
 
-    fn as_vtable(&self) -> *const *const Self::VTable {
+    #[inline(always)]
+    fn as_raw(&self) -> RawComPtr<Self> {
```

Comment:
I prefer this name - the name vtable is too confusing as it typically refers to the vtable layout rather than a particular vfptr.